newdle/client/src/components/ParticipantTablev2.js

```diff
@@ -0,0 +1,137 @@
+  timeslot: PropTypes.string,
+};
+
+function ParticipantRow({participant}) {
+  const timeslots = useSelector(getNewdleTimeslots);
+  let participantName = participant.name;
+  let participantEmail = participant.email;
+  // eslint-disable-next-line react/prop-types
+  let matches = participantName.match(/\b(\w)/g);
+  let initials = matches.join('');
+
+  return (
+    <Table.Row textAlign="center">
+      <Table.Cell>
+        <div style={{display: 'flex'}}>
+          <UserAvatar
+            user={{email: participantEmail, name: participantName, initials}}
+            size={25}
+            className={styles['participant-avatar']}
+          />
+          {participantName}
+        </div>
+      </Table.Cell>
+      {timeslots.map((timeslot, index) => {
+        let status = null;
+        if (participant.answers[timeslot] === 'available') {
+          status = 'available';
+        } else if (participant.answers[timeslot] === 'unavailable') {
+          status = 'unavailable';
+        } else if (participant.answers[timeslot] === 'ifneedbe') {
+          status = 'ifneedbe';
+        }
+        return (
+          <Table.Cell key={`${status}-${index}`} textAlign="center">
+            {!!status ? (
+              status === 'available' ? (
+                <Icon color="green" name="checkmark" size="large" />
+              ) : status === 'unavailable' ? (
+                <Icon color="red" name="close" size="large" />
+              ) : (
+                <Icon color="yellow" name="question" size="large" />
+              )
+            ) : null}
+          </Table.Cell>
+        );
+      })}
+    </Table.Row>
+  );
+}
+
+ParticipantRow.propTypes = {
+  participant: PropTypes.arrayOf(
+    PropTypes.shape({
+      answers: PropTypes.objectOf(PropTypes.oneOf(['unavailable', 'available', 'ifneedbe'])),
+      email: PropTypes.string,
+      name: PropTypes.string.isRequired,
+      status: PropTypes.string.isRequired,
+    })
+  ).isRequired,
+};
+
+export default function ParticipantTable() {
+  const timeslots = useSelector(getNewdleTimeslots);
+  const participants = useSelector(getNewdleParticipants);
+
+  if (timeslots.length === 0) {
+    return null;
+  }
+
+  return (
+    <div
+      style={{
+        overflowX: 'auto',
+        borderRight: '1px solid rgba(34, 36, 38, 0.15)',
+        borderLeft: '1px solid rgba(34, 36, 38, 0.15)',
+      }}
+    >
+      <Table textAlign="center">
+        <Table.Header>
+          <Table.Row>
+            <Table.HeaderCell
+              textAlign="center"
+              style={{paddingLeft: '40px', paddingRight: '40px'}}
+            >
+              Participants
+            </Table.HeaderCell>
+            {timeslots.map((timeslot, index) => (
+              <DateCell key={`${timeslot}-${index}`} timeslot={timeslot} />
+            ))}
+          </Table.Row>
+          {participants.map(participant => (
+            <ParticipantRow key={participant.name} participant={participant} />
```

Comment:
Participant rows are not sorted and can change when reloading. It would be great to have them somehow sorted (probably best alphabetically)!